core/eolearn/tests/test_core_tasks.py

```diff
@@ -148,28 +149,49 @@ def test_io_task_pickling(filesystem, task_class):
     assert isinstance(unpickled_task, task_class)
 
 
-def test_add_rename_remove_feature(patch):
-    cloud_mask = np.arange(10).reshape(5, 2, 1, 1)
-    feature_name = "CLOUD MASK"
-    new_feature_name = "CLM"
+@pytest.mark.parametrize(
+    "feature, feature_data",
+    [
+        ((FeatureType.MASK, "CLOUD MASK"), np.arange(10).reshape(5, 2, 1, 1)),
+        ((FeatureType.BBOX, None), BBox((24.54, 56.45, 95.4, 13.43), CRS(3857))),
+        ((FeatureType.TIMESTAMP, None), [datetime(2022, 1, 1, 10, 4, 7), datetime(2022, 1, 4, 10, 14, 5)]),
+    ],
+)
+def test_add_feature(feature: FeatureSpec, feature_data: np.ndarray) -> None:
+    # this test should fail for bbox and timestamps after rework
+    patch = EOPatch()
+    assert feature not in patch
+    patch = AddFeatureTask(feature)(patch, feature_data)
 
-    patch = copy.deepcopy(patch)
+    if isinstance(feature_data, np.ndarray):
+        assert_array_equal(patch[feature], feature_data)
+    else:
+        assert patch[feature] == feature_data
 
-    patch = AddFeatureTask((FeatureType.MASK, feature_name))(patch, cloud_mask)
-    assert np.array_equal(patch.mask[feature_name], cloud_mask), "Feature was not added"
 
-    patch = RenameFeatureTask((FeatureType.MASK, feature_name, new_feature_name))(patch)
-    assert np.array_equal(patch.mask[new_feature_name], cloud_mask), "Feature was not renamed"
-    assert feature_name not in patch[FeatureType.MASK], "Old feature still exists"
+def test_rename_feature(patch: EOPatch) -> None:
+    f_type, f_name, f_new_name = FeatureType.DATA, "bands", "new_bands"
+    assert (f_type, f_new_name) not in patch
+    patch_copy = copy.deepcopy(patch)
 
-    patch = RemoveFeatureTask((FeatureType.MASK, new_feature_name))(patch)
-    assert feature_name not in patch.mask, "Feature was not removed"
+    patch = RenameFeatureTask((f_type, f_name, f_new_name))(patch)
+    assert_array_equal(patch[(f_type, f_new_name)], patch_copy[(f_type, f_name)])
+    assert (f_type, f_name) not in patch
```

Comment:
add description, so that it is clear that the problem is that the input patch was changed

Reply:
done